Libs/MRML/Core/vtkMRMLSubjectHierarchyNode.cxx

```diff
@@ -1298,6 +1308,26 @@ void vtkSubjectHierarchyItem::SetUID(std::string uidName, std::string uidValue)
   this->Modified();
 }
 
+//---------------------------------------------------------------------------
+bool vtkSubjectHierarchyItem::RemoveUID(std::string uidName)
+{
+  if (this->UIDs.size() == 0)
+  {
+    return false;
+  }
+
+  // Use the find function to prevent adding an empty UID to the map
+  if (this->UIDs.find(uidName) != this->UIDs.end())
```

Comment:
maybe use iterator to avoid double lookup?
  if (auto it = this->UIDs.find(uidName); it != this->UIDs.end())
  {
    this->UIDs.erase(it);
    this->Modified();
    return true;
  }

Reply:
@sedrakSofthenge I agree with all your comments. I had to choose between consistency (with the existing code that does the same for item attributes) and potential performance gain. Since the performance gain is unlikely to be perceivable in any real-world scenarios (since subject hierarchy typically contains less than 20 items, and never more than a few hundred), I chose to go with less changes (and thus faster development and less chance of breaking anything). I'll see if I can spend a little more time with this to address some of your comments.